test/plog/PlogTest.cpp

```diff
@@ -0,0 +1,224 @@
+#include <k2/dto/MessageVerbs.h>
+#include <k2/appbase/AppEssentials.h>
+
+using namespace k2;
+
+PlogTest::PlogTest() {
+    K2INFO("ctor");
+}
+
+PlogTest::~PlogTest() {
+    K2INFO("dtor");
+}
+
+seastar::future<> PlogTest::gracefulStop() {
+    K2INFO("stop");
+    return std::move(_testFuture);
+}
+
+seastar::future<> PlogTest::start() {
+    K2INFO("start");
+    ConfigVar<String> configEp("cpo_url");
+    _cpoEndpoint = RPC().getTXEndpoint(configEp());
+
+    // let start() finish and then run the tests
+    _testTimer.set_callback([this] {
+        _testFuture = runTest2()
+        .then([this] { return runTest3(); })
+        .then([this] {
+            K2INFO("======= All tests passed ========");
+            exitcode = 0;
+        })
+        .handle_exception([this](auto exc) {
+            try {
+                std::rethrow_exception(exc);
+            } catch (RPCDispatcher::RequestTimeoutException& exc) {
+                K2ERROR("======= Test failed due to timeout ========");
+                exitcode = -1;
+            } catch (std::exception& e) {
+                K2ERROR("======= Test failed with exception [" << e.what() << "] ========");
+                exitcode = -1;
+            }
+        })
+        .finally([this] {
+            K2INFO("======= Test ended ========");
+            seastar::engine().exit(exitcode);
+        });
+    });
+    _testTimer.arm(0ms);
+    return seastar::make_ready_future<>();
+}
+
+seastar::future<> PlogTest::runTest1() {
+    K2INFO(">>> Test1: get non-existent persistence cluster");
+    auto request = dto::PersistenceClusterGetRequest{.name="Cluster1"};
+    return RPC()
+    .callRPC<dto::PersistenceClusterGetRequest, dto::PersistenceClusterGetResponse>(dto::Verbs::CPO_PERSISTENCE_CLUSTER_GET, request, *_cpoEndpoint, 100ms)
+    .then([](auto&& response) {
+        auto& [status, resp] = response;
+        K2EXPECT(status, Statuses::S404_Not_Found);
+    });
+}
+
+seastar::future<> PlogTest::runTest2() {
+    K2INFO(">>> Test2: create a persistence cluster");
+    dto::PersistenceGroup group1{.name="Group1", .plogServerEndpoints = _plogConfigEps()};
+    dto::PersistenceCluster cluster1;
+    cluster1.name="Cluster1";
+    cluster1.persistenceGroupVector.push_back(group1);
+
+    auto request = dto::PersistenceClusterCreateRequest{.cluster=std::move(cluster1)};
+    return RPC()
+    .callRPC<dto::PersistenceClusterCreateRequest, dto::PersistenceClusterCreateResponse>(dto::Verbs::CPO_PERSISTENCE_CLUSTER_CREATE, request, *_cpoEndpoint, 1s)
+    .then([](auto&& response) {
+        auto& [status, resp] = response;
+        K2EXPECT(status, Statuses::S201_Created);
+    });
+}
+
+seastar::future<> PlogTest::runTest3() {
+    K2INFO(">>> Test3: read the persistence group we created in test2");
+    return _client.getPersistenceCluster("Cluster1")
+    .then([this] () {
+        K2INFO("Test3.1: create a plog");
+        return _client.create();
+    })
+    .then([this] (auto&& response){
```

Comment:
you should add a test for double-creation (server should reject a create for existing plog)

Reply:
added